serialization/include/rp_ser.h

```diff
@@ -0,0 +1,401 @@
+};
+
+/**@brief Commands and events decoder structure. */
+struct rp_ser_decoders {
+	/** Command section start address. */
+	const struct rp_ser_cmd *cmd_begin;
+
+	/** Command section end address. */
+	const struct rp_ser_cmd *cmd_end;
+
+	/** Event section start address. */
+	const struct rp_ser_evt *evt_begin;
+
+	/** Event section end address. */
+	const struct rp_ser_evt *evt_end;
+};
+
+/**@brief Helper macro for creating command decoder. All comands decoders have to be assigned
+ *        to proper Remote Procedure Serialization instance. After receiving a command, the
+ *        command decoder function is searching in command data memory section based on command
+ *        number and calls the actual decoder.
+ *
+ * @param[in] _rp_inst Remote Procedure Serialization instance.
+ * @param[in] _name Command decoder name.
+ * @param[in] _cmd Command number.
+ * @param[in] _handler Command decoder function @ref cmd_handler_t.
+ */
+#define RP_SER_CMD_DECODER(_rp_inst, _name, _cmd, _handler)         \
+	RP_STATIC_ASSERT(&_rp_inst != NULL, "Invalid instance");    \
+	RP_STATIC_ASSERT(_cmd <= 0xFF, "Command out of range");     \
+	const struct rp_ser_cmd RP_CONCAT(_name, _cmd_dec) __used   \
+	__attribute__((__section__("." RP_STRINGIFY(rp_ser_decoder) \
+				   "." "cmd"                        \
+				   "." RP_STRINGIFY(_rp_inst)       \
+				   "." RP_STRINGIFY(_name)))) = {   \
+		.cmd = _cmd,					    \
+		.func = _handler				    \
+	}
+
+/**@brief Helper macro for creating event decoder. All events decoders have to be assigned
+ *        to proper Remote Procedure Serialization instance. After receiving a event the
+ *        event decoder function is searching in command data memory section based on command
+ *        number and called.
+ *
+ * @param[in] _rp_inst Remote Procedure Serialization instance.
+ * @param[in] _name Event decoder name.
+ * @param[in] _cmd Event number.
+ * @param[in] _handler Event decoder function @ref cmd_handler_t.
+ */
+#define RP_SER_EVT_DECODER(_rp_inst, _name, _evt, _handler)         \
+	RP_STATIC_ASSERT(&_rp_inst != NULL, "Invalid instance");    \
+	RP_STATIC_ASSERT(_evt <= 0xFF, "Event out of range");       \
+	const struct rp_ser_evt RP_CONCAT(_name, _evt_dec) __used   \
+	__attribute__((__section__("." RP_STRINGIFY(rp_ser_decoder) \
+				   "." "evt"                        \
+				   "." RP_STRINGIFY(_rp_inst)       \
+				   "." RP_STRINGIFY(_name)))) = {   \
+		.evt = _evt                                         \
+		.func = _handler			            \
+	}
+
+/**@brief Endpoint configuration structure. */
+struct rp_ser_endpoint {
+	/** Endpoint number. */
+	int number;
+};
+
+/**@brief Remote Procedure Serialization instance. */
+struct rp_ser {
+	/** Transport endpoint instance. */
+	struct rp_trans_endpoint endpoint;
+
+	/** Decoders section addresses. */
+	const struct rp_ser_decoders *decoders;
+
+	/** Transport endpoint initial configuration. */
+	const struct rp_ser_endpoint *ep_conf;
```

Comment:
why does the initial endpoint configuration have to be inside serialization instance?